src/protocols/LoRaWAN/LoRaWAN.cpp

```diff
@@ -48,172 +44,208 @@ void LoRaWANNode::setCSMA(uint8_t backoffMax, uint8_t difsSlots, bool enableCSMA
-  execMacCommand(&cmd, false);
+  memcpy(cmd.payload, &this->bufferSession[RADIOLIB_LORAWAN_SESSION_REJOIN_PARAM_SETUP], cmd.len);
+  execMacCommand(&cmd);
 
-  uint8_t queueBuff[sizeof(LoRaWANMacCommandQueue_t)] = { 0 };
-  mod->hal->readPersistentStorage(mod->hal->getPersistentAddr(RADIOLIB_EEPROM_LORAWAN_MAC_QUEUE_UL_ID), queueBuff, sizeof(LoRaWANMacCommandQueue_t));
-  memcpy(&this->commandsUp, queueBuff, sizeof(LoRaWANMacCommandQueue_t));
-  RADIOLIB_DEBUG_PROTOCOL_PRINTLN("Number of MAC commands: %d", this->commandsUp.numCommands);
+  // copy uplink MAC command queue back in place
+  memcpy(&this->commandsUp, &this->bufferSession[RADIOLIB_LORAWAN_SESSION_MAC_QUEUE_UL], sizeof(LoRaWANMacCommandQueue_t));
 
   state = this->setPhyProperties();
   RADIOLIB_ASSERT(state);
 
   // full session is restored, so set joined flag to whichever mode is restored
-  this->activeMode = lwMode;
-
-  return(this->activeMode);
-}
-
-void LoRaWANNode::restoreFcntUp() {
-  Module* mod = this->phyLayer->getMod();
-
-  uint8_t fcntBuffStart = mod->hal->getPersistentAddr(RADIOLIB_EEPROM_LORAWAN_FCNT_UP_ID);
-  uint8_t fcntBuffEnd = mod->hal->getPersistentAddr(RADIOLIB_EEPROM_LORAWAN_FCNT_UP_ID + 1);
-  uint8_t buffSize = fcntBuffEnd - fcntBuffStart;
-  #if RADIOLIB_STATIC_ONLY
-  uint8_t fcntBuff[RADIOLIB_STATIC_ARRAY_SIZE];
-  #else
-  uint8_t* fcntBuff = new uint8_t[buffSize];
-  #endif
-  mod->hal->readPersistentStorage(mod->hal->getPersistentAddr(RADIOLIB_EEPROM_LORAWAN_FCNT_UP_ID), fcntBuff, buffSize);
-
-  // copy the two most significant bytes from the first two bytes
-  uint32_t bits_30_22 = (uint32_t)fcntBuff[0];
-  uint32_t bits_22_14 = (uint32_t)fcntBuff[1];
-
-  // the next 7 bits must be retrieved from the byte to which was written most recently
-  // this is the last byte that has its state bit (most significant bit) set equal to its predecessor
-  // we find the first byte that has its state bit different, and subtract one
-  uint8_t idx = 2;
-  uint8_t state = fcntBuff[idx] >> 7;
-  for(; idx < 5; idx++) {
-    if(fcntBuff[idx] >> 7 != state) {
-      break;
-    }
-  }
-  uint32_t bits_14_7 = (uint32_t)fcntBuff[idx-1] & 0x7F;
-
-  // equally, the last 7 bits must be retrieved from the byte to which was written most recently
-  // this is the last byte that has its state bit (most significant bit) set equal to its predecessor
-  // we find the first byte that has its state bit different, and subtract one
-  idx = 5;
-  state = fcntBuff[idx] >> 7;
-  for(; idx < buffSize; idx++) {
-    if(fcntBuff[idx] >> 7 != state) {
-      break;
-    }
-  }
-  uint32_t bits_7_0 = (uint32_t)fcntBuff[idx-1] & 0x7F;
-  #if !RADIOLIB_STATIC_ONLY
-  delete[] fcntBuff;
-  #endif
+  this->activeMode = LoRaWANNode::ntoh<uint16_t>(&this->bufferNonces[RADIOLIB_LORAWAN_NONCES_MODE]);
 
-  this->fcntUp = (bits_30_22 << 22) | (bits_22_14 << 14) | (bits_14_7 << 7) | bits_7_0;
+  return(RADIOLIB_ERR_NONE);
```

Comment:
This should return `state` - I do realize that the only way for the program to get to here is if `state == RADIOLIB_ERR_NONE`, however, that could change in the future, even if by mistake.